### DISCUSSION TITLE
Issue: "JFrog's Free-Tier subscription has been deactivated."

### DISCUSSION THREAD

**Discussion by udippel** [gerbera/gerbera#2967] [General]
... this is what I get when I want to download the most recent build at the usual URL.
Any changes?

**Comment by KarlStraussberger** [comment]
@whyman Did you get any notice from them or do you have to renew our free subscription

**Comment by whyman** [comment]
Good question! I will check my email. I don't recall seeing anything though 😞

**Reply by whyman** [reply]
Nothing I could see. At the very least terrible communication.

I think we need to find somewhere else to host the repos 😭

**Reply by tommyp-vr4** [reply]
I'm not an expert on making packages, but there is what I think is called a buildbot at https://build.opensuse.org/. There is a browser package I use from there that is built for ubuntu.

**Reply by KarlStraussberger** [reply]
I know opensuse, but don't have time to work out a build setup for them.

**Reply by whyman** [reply]
Just popping in to say I haven't forgotten about this. Time is just not available (hopefully I'm a few weeks!)

**Comment by gobater** [comment]
@KarlStraussberger , @whyman , is this just a hosting issue? If that's the case, couldn't we add some of the pre-build packages to github as part of the release artifacts (not sure about the limitations). That would definitely help to keep spreading Gerbera to non-experts.

Nevertheless, maybe an option would be the provide Docker containers and instructions to build the packages for most commonly used platforms for those willing to run Gerbera natively (without docker).

**Reply by KarlStraussberger** [reply]
It's the indexing that somehow provided for `apt`. I'm not really an expert in ubuntu/debian.

All the rest is just a question of time or someone who exactly know what to do.

**Reply by gobater** [reply]
You can install packages in ubuntu/debian without any "index" (apt compatible source repository) given you have the required DEB package (and dependencies)...

For ubuntu, the problem could be probably solved using launchpad, but I'm not sure how one could manage own dependencies (e.g. newer versions of libupnp than the provided in the official ubuntu repositories) with launchpad. Maybe it's possible to add them (also as DEB) into the same PPA.

I imagine that the packages provided in the the former jfrog repository where build with a kind of CI with Docker + some scripts. It's the usual way to build packages for several platforms (e.g. ubuntu 20, ubuntu 22, raspberry/debian bullseye, raspberry/debian bookworkm, ...)

I'm also not an expert in debian package management but willing to support in everything I can (even if that means building packages locally and uploading them somewhere...)

**Comment by udippel** [comment]
I'd be most happy if JFrog returned. Very egoistic: I loved to follow up with all iterations. 
Also to help out, because over the last months I could identify some bugs creeping in through further development. I can't build myself, don't want to; just seeing me as 'end user tester'. 
Plus, with various distributions, including some ARM devices an advanced automated build-system is to be preferred.

**Reply by gobater** [reply]
Where there's a will there's a way...
"Automated build-system" is always best, no doubt of it. You can build for ARM on x86 machines, in case you never heard about cross-compiling before.

If you want JFrog back, it's quite easy. You can pay for it :-)

**Reply by udippel** [reply]
I don't want to spoil your weekend by including the pay slip of my retirement benefits. I promise, I'll come back the very moment when the next 7-digit-win in the lottery comes in.

**Comment by udippel** [comment]
Only in case you haven't noticed yet, JFrog changed its display. Now it stands on "Reactivation failed". Not good.

**Comment by whyman** [comment]
This is now resolved, please update your configs per the docs https://docs.gerbera.io/en/stable/install.html#ubuntu-mint

**Comment by udippel** [comment]
I wouldn't bet on it. Just going to my previous URL where I used to go and scroll down until the most recent version popped up, still gives me the 'landing error'.

Well, I might have to actually follow the instructions:

```
curl -fsSL https://gerbera.jfrog.io/artifactory/api/gpg/key/public | sudo apt-key add -
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
gpg: no valid OpenPGP data found.
$ sudo apt-add-repository https://gerbera.jfrog.io/artifactory/debian-git
Repository: 'deb https://gerbera.jfrog.io/artifactory/debian-git jammy main'
Description:
Archive for codename: jammy components: main
More info: https://gerbera.jfrog.io/artifactory/debian-git
Adding repository.
Press [ENTER] to continue or Ctrl-c to cancel.
Adding deb entry to /etc/apt/sources.list.d/archive_uri-https_gerbera_jfrog_io_artifactory_debian-git-jammy.list
Adding disabled deb-src entry to /etc/apt/sources.list.d/archive_uri-https_gerbera_jfrog_io_artifactory_debian-git-jammy.list
[...]
Err:3 https://gerbera.jfrog.io/artifactory/debian-git jammy InRelease
  Clearsigned file isn't valid, got 'NOSPLIT' (does the network require authentication?)
[...]
Reading package lists... Done                            
E: Failed to fetch https://gerbera.jfrog.io/artifactory/debian-git/dists/jammy/InRelease  Clearsigned file isn't valid, got 'NOSPLIT' (does the network require authentication?)
E: The repository 'https://gerbera.jfrog.io/artifactory/debian-git jammy InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
$ 
```
My personal preferred way would still be pointing the browser to the URL, and see, and select, all versions for all platforms, and select the one that I desire. I did need that in the past, too, when the most recent version had a bug, and I went backwards to see with which intermittent version that bug had been introduced.

**Reply by Martii** [reply]
@udippel Cc: @whyman 

The GitHub *(GH)* repo is fixed  however, even for me, it appears that [the documentation host site](https://docs.gerbera.io/en/stable/install.html#ubuntu-mint) was/is not currently synchronized with the "non-artifactory" urls.

**Reply by whyman** [reply]
Ah yes, apologies you will need to see the "[latest](https://docs.gerbera.io/en/latest/install.html#ubuntu-mint)" version as the stable will not get updated until the next release is tagged.

**Comment by udippel** [comment]
Tried that one. I actually had expected something like this, but hadn't been sure. 
In any case, goes through, though leaves some warnings behind:

```
$ cat logs/debug_v2.1.0-7-gb8cf9248 
[...]
2024-04-14 00:13:31   info: IPv6 ULA/GLA: Server bound to: :0
2024-04-14 00:13:31 warning: Rejected attempt to load host '192.168.1.2:49152' as it does not match configured virtualURL/externalURL: ''/''. See https://docs.gerbera.io/en/stable/config-server.html#virtualurl
2024-04-14 00:13:31 warning: Rejected attempt to load host '192.168.1.2:49152' as it does not match configured virtualURL/externalURL: ''/''. See https://docs.gerbera.io/en/stable/config-server.html#virtualurl
2024-04-14 00:13:31 warning: Rejected attempt to load host '192.168.1.2:49152' as it does not match configured virtualURL/externalURL: ''/''. See https://docs.gerbera.io/en/stable/config-server.html#virtualurl

```

**Comment by udippel** [comment]
The URL ([https://pkg.gerbera.io/*]) gives:
"Error 404
This object could not be viewed
You are not authorized to view this object

This object does not exist or is not publicly accessible at this URL. Check the URL of the object that you're looking for or contact the owner to enable Public access."

I don't consider this favourable. Had pointed out earlier and above that it makes a lot of sense to offer any version as .deb at any time beyond the latest update. Otherwise everything is lost in the internals of apt, instead of some ~/Sources/gerbera/ to which one could download a .deb; and manually up- or downgrade.

**Reply by whyman** [reply]
Yeah we will address this at some point, being able to browse is useful
